markdown/extensions/toc.py

```diff
@@ -236,8 +235,12 @@ def run(self, doc):
         for el in doc.iter():
             if isinstance(el.tag, string_type) and self.header_rgx.match(el.tag):
                 self.set_level(el)
-                if int(el.tag[-1]) < int(self.toc_height) or int(el.tag[-1]) > int(self.toc_depth):
-
+                if type( self.toc_depth ) != int:
```

Comment:
Note that `self.toc_depth` could reasonably be a string which contains a single integer. Testing for an `int` here will fail in many cases. You probably need to do something like:

```python
try:
    toc_bottom = int(self.toc_depth)
    toc_top = 1
except ValueError:
    toc_top, toc_bottom = [int(t), int(b) for t, b in self.toc_depth.split('-')]
```

And that last line should perhaps be wrapped in its own try/except block to catch and raise an error if a bad range is provided.